src/runtime-rs/crates/runtimes/virt_container/src/container_manager/manager.rs

```diff
@@ -57,12 +63,37 @@ impl ContainerManager for VirtContainerManager {
     async fn create_container(&self, config: ContainerConfig, spec: oci::Spec) -> Result<PID> {
         let container = Container::new(
             self.pid,
-            config,
+            config.clone(),
+            spec.clone(),
             self.agent.clone(),
             self.resource_manager.clone(),
         )
         .context("new container")?;
 
+        // CreateContainer Hooks:
+        // * should be run in vmm namespace (hook path in runtime namespace)
+        // * should be run after the vm is started, before container is created, and after CreateRuntime Hooks
+        // * spec details: https://github.com/opencontainers/runtime-spec/blob/c1662686cff159595277b79322d0272f5182941b/config.md#createcontainer-hooks
+        let vmm_master_tid = self.hypervisor.get_vmm_master_tid().await?;
```

Comment:
Please add a helper function to run the hook so that we don't have to maintain the same piece of code at different places.

Reply:
I used to think the `hookstates.execute_hooks(...)` is already the helper function, since we should construct different oci::State for different kinds of hooks. I wonder, do we need to consider the hook type in the helper function? By that way, we should construct different oci::State according the hook type in that function?